### ISSUE TITLE
Section width

### ISSUE THREAD

**Issue by Alex293** [devxoul/UICollectionViewFlexLayout#4]
Hi, nice layout !

I may have missed something but the section wrap to the width of the items. So if the items don't take the whole collection width the section won't either. This is kinda weird as I was expecting the section width to be defined by margins / paddings set on the delegate relatively to the view.

**Comment by Alex293** [comment]
Here is a screen shot : 
<img width="330" alt="screen shot 2018-06-09 at 16 44 55" src="https://user-images.githubusercontent.com/5930273/41192826-772529d0-6c04-11e8-8e27-da3e7382a4f8.png">

and the code:

```swift
extension DashboardScreen: UICollectionViewDelegateFlexLayout {
	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewFlexLayout, verticalSpacingBetweenSectionAt section: Int, and nextSection: Int) -> CGFloat {
		return 20
	}
	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewFlexLayout, marginForSectionAt section: Int) -> UIEdgeInsets {
		return UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
	}
	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewFlexLayout, paddingForSectionAt section: Int) -> UIEdgeInsets {
		return UIEdgeInsets(top: 16, left: 16, bottom: 16, right: 16)
	}

	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewFlexLayout, horizontalSpacingBetweenItemAt indexPath: IndexPath, and nextIndexPath: IndexPath) -> CGFloat {
		return 8
	}

	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewFlexLayout, verticalSpacingBetweenItemAt indexPath: IndexPath, and nextIndexPath: IndexPath) -> CGFloat {
		return 8
	}

	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewFlexLayout, marginForItemAt indexPath: IndexPath) -> UIEdgeInsets {
		return UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
	}

	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewFlexLayout, paddingForItemAt indexPath: IndexPath) -> UIEdgeInsets {
		return UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
	}

	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewFlexLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
		return CGSize(width: 50, height: 200)
	}

	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewFlexLayout, zIndexForItemAt indexPath: IndexPath) -> Int {
		return 2
	}
}
```

```swift
extension DashboardScreen: UICollectionViewDataSource {

	func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
		return items[section].count
	}

	func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
		return collectionView.dequeueReusableCell(withReuseIdentifier: DashboardScreenCollectionCell.identifier, for: indexPath)
	}

	func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
		switch kind {
		case DashboardScreenCollectionSectionBackground.kind:
			return collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: DashboardScreenCollectionSectionBackground.identifier, for: indexPath)
		case DashboardScreenCollectionItemBackground.kind:
			return collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: DashboardScreenCollectionItemBackground.identifier, for: indexPath)
		default:
			fatalError("Unmanaged kind: \(kind)")
		}
	}

	func numberOfSections(in collectionView: UICollectionView) -> Int {
		return items.count
	}
}
```

```swift
func setupCollectionView() {
		let collectionView = UICollectionView(frame: view.frame, collectionViewLayout: UICollectionViewFlowLayout())
		self.collectionView = collectionView
		collectionView.translatesAutoresizingMaskIntoConstraints = false
		view.addSubview(collectionView)
		if #available(iOS 11.0, *) {
			collectionView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 8).isActive = true
			collectionView.leftAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leftAnchor, constant: 8).isActive = true
			collectionView.rightAnchor.constraint(equalTo: view.safeAreaLayoutGuide.rightAnchor, constant: -8).isActive = true
			collectionView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -8).isActive = true
		} else {
			collectionView.topAnchor.constraint(equalTo: view.topAnchor, constant: 8).isActive = true
			collectionView.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 8).isActive = true
			collectionView.rightAnchor.constraint(equalTo: view.rightAnchor, constant: -8).isActive = true
			collectionView.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: 8).isActive = true
		}
		collectionView.backgroundColor = .clear
		collectionView.register(DashboardScreenCollectionCell.self, forCellWithReuseIdentifier: DashboardScreenCollectionCell.identifier)
		collectionView.register(DashboardScreenCollectionSectionBackground.self, forSupplementaryViewOfKind: DashboardScreenCollectionSectionBackground.kind, withReuseIdentifier: DashboardScreenCollectionSectionBackground.identifier)
		collectionView.register(DashboardScreenCollectionItemBackground.self, forSupplementaryViewOfKind: DashboardScreenCollectionItemBackground.kind, withReuseIdentifier: DashboardScreenCollectionItemBackground.identifier)
		collectionView.dataSource = self
		let layout = UICollectionViewFlexLayout()
		collectionView.delegate = self
		collectionView.collectionViewLayout = layout
	}
```

Item & section are dumb class for now

**Comment by devxoul** [comment]
Yeah it can be confused. #5 is a patch for this issue. I'm worried that this PR is a breaking change, so I'd be glad if you could test in your own project too. Thanks!

**Comment by Alex293** [comment]
Nice as the project wasn't updated since 7 months I didn't expected to get an answer as quickly :)
I made some changes in my project to achieve the same thing. I'll compare the two and give you feedback tomorrow if I can.

**Comment by Alex293** [comment]
Also now that the size is fixed, it might be nice to center the elements or maybe as an option ? What do you think ? If you don't like it I'll go with large padding.

**Comment by Alex293** [comment]
Your branch is working for my use case

**Comment by devxoul** [comment]
@Alex293, Thanks for check. I released [1.1.0](https://github.com/devxoul/UICollectionViewFlexLayout/releases/tag/1.1.0) which contains the patch.

Actually the section alignment was one of my roadmap but removed it because I didn't need that feature at that time. I think I can consider implementing section alignment when I have spare time.

**Comment by Alex293** [comment]
Great !
